lib/tools/adb-commands.js

```diff
@@ -965,11 +970,26 @@ methods.isDataOn = async function isDataOn () {
  *                   is not equal to '0.0'.
  */
 methods.isAnimationOn = async function isAnimationOn () {
-  let animator_duration_scale = await this.getSetting('global', 'animator_duration_scale');
-  let transition_animation_scale = await this.getSetting('global', 'transition_animation_scale');
-  let window_animation_scale = await this.getSetting('global', 'window_animation_scale');
-  return _.some([animator_duration_scale, transition_animation_scale, window_animation_scale],
-                (setting) => setting !== '0.0');
+  return (await B.all(ANIMATION_SCALE_KEYS.map(
+    async (k) => (await this.getSetting('global', k)) !== '0.0'))
+  ).includes(true);
+};
+
+/**
+ * Set animation scale with the given velue.
+ *
+ * @this {import('../adb.js').ADB}
+ * @param {number} value Animation value (int or floating) to set. Negative value sets zero.
```

Comment:
I am still confused as of what this value means.
What is the acceptable value range? What is the effect of increasing/decresing it? What is the minimum/maximum value and how they affect the actual device UI?

Reply:
Hm, let me check the max by reading their code. I could set `100000000000000000000000000000000` for example but the animation in the UI was very slow so it looks hang (but that was working). negative values are handled as off (zero) with the settings command.

Reply:
Maybe around https://cs.android.com/android/platform/superproject/+/master:packages/apps/Settings/src/com/android/settings/development/TransitionAnimationScalePreferenceController.java;l=92?q=transition_animation_scale&ss=android&start=1 , so the max is Java's float. Let me address it

Reply:
h, over the float value can be set as the settings command, so the float max is not max as the adb shell settings.

Reply:
One more thing is that the settings's UI showed an "off" animation after the `adb shell settings put global transition_animation_scale 100`, but the actual behavior was that the transition was very slow compared to 10. I reboot the device after setting the value as 1 as kind of endless animation with large value, btw.